x-pack/plugins/ml/public/application/timeseriesexplorer/timeseriesexplorer_utils/timeseriesexplorer_utils.js

```diff
@@ -37,218 +29,6 @@ export function createTimeSeriesJobData(jobs) {
-          if (causes.length > 0) {
-            chartPoint.byFieldName = record.by_field_name;
-            chartPoint.numberOfCauses = causes.length;
-            if (causes.length === 1) {
-              // If only a single cause, copy actual and typical values to the top level.
-              const cause = record.causes[0];
-              chartPoint.actual = cause.actual;
-              chartPoint.typical = cause.typical;
-              // substitute the value with the record's actual so it won't plot as null/0
-              if (chartPoint.value === null) {
-                chartPoint.value = cause.actual;
-              }
-            }
-          }
-        }
-
-        if (
-          record.anomaly_score_explanation !== undefined &&
-          record.anomaly_score_explanation.multi_bucket_impact !== undefined
-        ) {
-          chartPoint.multiBucketImpact = record.anomaly_score_explanation.multi_bucket_impact;
-        }
-
-        chartPoint.isMultiBucketAnomaly = isMultiBucketAnomaly(record);
-      }
-    }
-  });
-
-  return chartData;
-}
-
-// Adds a scheduledEvents property to any points in the chart data set
-// which correspond to times of scheduled events for the job.
-export function processScheduledEventsForChart(chartData, scheduledEvents, aggregationInterval) {
-  if (scheduledEvents !== undefined) {
-    const timesToAddPointsFor = [];
-
-    // Iterate through the scheduled events making sure we have a chart point for each event.
-    const intervalMs = aggregationInterval.asMilliseconds();
-    let lastChartDataPointTime = undefined;
-    if (chartData !== undefined && chartData.length > 0) {
-      lastChartDataPointTime = chartData[chartData.length - 1].date.getTime();
-    }
-
-    // In case there's no chart data/sparse data during these scheduled events
-    // ensure we add chart points at every aggregation interval for these scheduled events.
-    let sortRequired = false;
-    each(scheduledEvents, (events, time) => {
-      const exactChartPoint = findChartPointForScheduledEvent(chartData, +time);
-
-      if (exactChartPoint !== undefined) {
-        exactChartPoint.scheduledEvents = events;
-      } else {
-        const timeToAdd = Math.floor(time / intervalMs) * intervalMs;
-        if (timesToAddPointsFor.indexOf(timeToAdd) === -1 && timeToAdd !== lastChartDataPointTime) {
-          const pointToAdd = {
-            date: new Date(timeToAdd),
-            value: null,
-            scheduledEvents: events,
-          };
-
-          chartData.push(pointToAdd);
-          sortRequired = true;
-        }
-      }
-    });
-
-    // Sort chart data by time if extra points were added at the end of the array for scheduled events.
-    if (sortRequired === true) {
-      chartData.sort((a, b) => a.date.getTime() - b.date.getTime());
-    }
-  }
-
-  return chartData;
-}
-
 // Finds the chart point which is closest in time to the specified time.
 export function findNearestChartPointToTime(chartData, time) {
```

Comment:
We can remove this file if this last function `findNearestChartPointToTime` isn't needed - I couldn't find it being used anywhere but wanted to get a second opinion before just removing it. 
cc @walterra, @jgowdyelastic

Reply:
I think this used to be for locating closest points in the source time series for adding anomaly markers when the chart agg interval > bucket span. But as you say, I can't see it being used anywhere now.

Reply:
Removed in e4d02f9bf8684351d1b6c99eac225b9557cb2c96